quantum/quantum_keycodes.h

```diff
@@ -96,6 +96,7 @@ enum quantum_keycodes {
     DEBUG,
     MAGIC_SWAP_CONTROL_CAPSLOCK,
     MAGIC_CAPSLOCK_TO_CONTROL,
+    MAGIC_SWAP_CONTROL_LGUI,
```

Comment:
Can new keycodes be added to the end to maintain VIA compatibility?

Reply:
I would much rather that related keycodes stay grouped together instead of strewn all over the place.

And just quietly, requiring these enum values to be constant defeats the entire purpose of the enum. Some of them are already not known because they are surrounded by `#ifdef`s. The keycode space really needs some proper allocation...

Reply:
Yes, and I was told this "refactoring" was going to happen, and until then, new keycodes added to the end, or at least batched together into one breaking commit.

Reply:
I suggest moving it to the end for now and later for a breaking release it can be moved back to the logical spot?

Reply:
Yeah, I also would prefer the enums grouped together by function, I'm not against that principle... just asking for a short term reprieve from VIA having to deal with the issue of "versioned" QMK keycode name -> int value mappings, we're in the middle of refactoring and adding other stuff and would really like to get to a release of that first before tackling others. Beyond the scope of this PR though, will discuss elsewhere.